sdk/apis/apis/v1alpha2/types_apiexport_conversion_test.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	apisv1alpha1 "github.com/kcp-dev/kcp/sdk/apis/apis/v1alpha1"
+)
+
+func TestConvertV1Alpha2APIExports(t *testing.T) {
+	testcases := []APIExport{
+		{
+			Spec: APIExportSpec{},
+		},
+		{
+			Spec: APIExportSpec{
+				ResourceSchemas: []ResourceSchema{{
+					Schema: "foo",
+					Storage: ResourceSchemaStorage{
+						CRD: &ResourceSchemaStorageCRD{},
+					},
+				}},
+			},
+		},
+		{
+			Spec: APIExportSpec{
+				ResourceSchemas: []ResourceSchema{{
+					Schema: "foo",
+					Storage: ResourceSchemaStorage{
+						CRD: &ResourceSchemaStorageCRD{},
+					},
+				}, {
+					Schema: "bar",
+					Storage: ResourceSchemaStorage{
+						CRD: &ResourceSchemaStorageCRD{},
+					},
+				}},
+			},
+		},
+		{
+			Spec: APIExportSpec{
+				ResourceSchemas: []ResourceSchema{{
+					Schema: "bar",
+					Storage: ResourceSchemaStorage{
+						CRD: &ResourceSchemaStorageCRD{},
+					},
+				}, {
+					Schema: "foo",
+					Storage: ResourceSchemaStorage{
+						Virtual: &ResourceSchemaStorageVirtual{},
+					},
+				}},
+			},
+		},
+		{
+			Spec: APIExportSpec{
+				ResourceSchemas: []ResourceSchema{{
+					Schema: "foo",
+					Storage: ResourceSchemaStorage{
+						Virtual: &ResourceSchemaStorageVirtual{
+							Reference: corev1.TypedLocalObjectReference{
+								Kind: "test",
+							},
+						},
+					},
+				}},
+			},
```

Comment:
how do we know that everything in the spec is copied over to v1alpha1 and back? This would be tested with fuzzing. We must have these tests. Somebody adds a field and forgets the conversion. Then we have dataloss 😞